synthesizer/program/src/logic/instruction/operation/cast.rs

```diff
@@ -63,7 +66,9 @@ impl<N: Network> Parser for CastType<N> {
         alt((
             map(tag("group.x"), |_| Self::GroupXCoordinate),
             map(tag("group.y"), |_| Self::GroupYCoordinate),
-            map(RegisterType::parse, |register_type| Self::RegisterType(register_type)),
+            map(pair(Locator::parse, tag(".record")), |(locator, _)| Self::ExternalRecord(locator)),
+            map(pair(Identifier::parse, tag(".record")), |(identifier, _)| Self::Record(identifier)),
+            map(PlaintextType::parse, |plaintext_type| Self::Plaintext(plaintext_type)),
```

Comment:
```suggestion
            map(PlaintextType::parse, Self::Plaintext),
```
pretty sure this can be simplified in this fashion